client/a8c-for-agencies/components/pending-payment-notification/index.tsx

```diff
@@ -1,54 +1,89 @@
 import { Button } from '@wordpress/components';
 import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
 import LayoutBanner from 'calypso/a8c-for-agencies/components/layout/banner';
-import { useDispatch, useSelector } from 'calypso/state';
-import { getActiveAgency } from 'calypso/state/a8c-for-agencies/agency/selectors';
-import { setPreference } from 'calypso/state/preferences/actions';
-import { getPreference } from 'calypso/state/preferences/selectors';
+import useFetchInvoices from 'calypso/a8c-for-agencies/data/purchases/use-fetch-invoices';
 import { CONTACT_URL_HASH_FRAGMENT } from '../a4a-contact-support-widget';
 import { A4A_INVOICES_LINK } from '../sidebar-menu/lib/constants';
 
 import './style.scss';
 
-const PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE = 'pending-payment-notification-dismissed';
+// TODO: Uncomment this if we would want to dismiss the notification.
+// const PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE = 'pending-payment-notification-dismissed';
 
 export default function PendingPaymentNotification() {
-	const dispatch = useDispatch();
-	const agency = useSelector( getActiveAgency );
-
-	const isDismissed = useSelector( ( state ) =>
-		getPreference( state, PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE )
-	);
-
-	const canIssueLicenses = agency?.can_issue_licenses ?? true;
+	const invoices = useFetchInvoices( { starting_after: '', ending_before: '' }, undefined, 'paid' );
 
 	const translate = useTranslate();
 
-	// If the agency can issue licenses or the notification has been dismissed, do not show the notification.
-	if ( canIssueLicenses || isDismissed ) {
+	const daysDue = useMemo( () => {
+		const latestUnpaidInvoice =
```

Comment:
I think this is getting the oldest invoice rather than the latest (newest), right? If so maybe we could adjust the name to make that clear?
_I think_ looking at the oldest is right, is that the intention?

Reply:
Yeah, I want the oldest here, you're right. It's wrong naming.